chore: add docz (#3)
leroy-merlin-br/backyard-ui: docs/documenting/documentation.mdx

---
name: Documenting
route: /documenting
---

# Documenting

> When creating a new component in this library, you need to create documentation for this component. If you can not, your _Pull Request_ will be disapproved by the team that will review your code.

Reviewer: ```suggestion
When creating a new component in this library, you must document it. If you don't create the documentation for your component, your _Pull Request_ will be rejected by the reviewers.
```

> Therefore, the following will be an example of how the documentation of the new components should be made to follow the pattern of development.

Reviewer: ```suggestion
Therefore, the following is an example of how to make documentation of new components, following this library standards.
```

## Topics

List of all topics in this document.

- [Introduction](#introduction)
- [Starting](#starting)
- [Structure of folders](#structure-of-folders)
- [How to use](#how-to-use)

> - [Props](#props)

Reviewer: this link is broken

## Introduction

We encourage our developers to always create documentation of our components to facilitate the use of the user that will consume it.

> The purpose of this document is to show you how to proceed after you implement your component. The idea is that when you reach the end of this reading you will be able to follow our documentation standards.

Reviewer: ```suggestion
The purpose of this document is to show to you how to proceed after you implement your component. The idea is that when you reach the end of this reading you will be able to follow our documentation standards.
```

## Starting

> Follow the steps below to begin building your component documentation.

Reviewer: I don't think that we need this section, it is redundant.

Author: Ok, i'm will remove this section from document.

## Structure of folders

> We have a mass structure for the components so we can follow them.

Reviewer: I did not understand what this phrase means.

We hope you have a structure similar to this one:

![base](./images/base-structure.png)

You should create a folder named `docs` and inside it will be where the documentation of your component should be included. Simple components may contain only one file, more complex components may contain more than one file. But in cases of more extensive or complex components, we advise you to keep only one file in the folder and separate by topics to keep the rest of the documentation standard and also for organizational issues.

In the end we expect its structure to be this way:

![base](./images/base-structure-with-doc.png)

## How to use

> We hope that the documentation for your component contains some topics to facilitate the use and understanding of the user that will consume this component.

Reviewer: This phrase looks a lot like the one in introduction.

> Below the list of topics to be documented:

Reviewer: ```suggestion
List of topics to be covered:
```

- **How to use**
  - This includes details of how the component should be called, in which cases it makes sense, and so on ...
- **Variations**

>   - If your component has variations, documentation should contemplate these variations available.

Reviewer: ```suggestion
  - If your component has variations, the documentation should contemplate them.
```

- **Examples**
  - This section should contain examples of how your component renders on the screen.
- **Props**
  - This section should be directed only to document the `API` available in your component. It should be created in tabular form with information about the available properties.

> ## Example

Reviewer: I'm lost with this example.

Author: i'm will remove this.
